clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -18410,6 +18410,24 @@ Value *CodeGenFunction::EmitAMDGPUBuiltinExpr(unsigned BuiltinID,
         CGM.getIntrinsic(Intrinsic::amdgcn_update_dpp, Args[0]->getType());
     return Builder.CreateCall(F, Args);
   }
+  case AMDGPU::BI__builtin_amdgcn_readlane:
+  case AMDGPU::BI__builtin_amdgcn_readfirstlane: {
```

Comment:
readfirstlane case can just call emitUnaryBuiltin. readlane can probably get away with emitBinaryBuiltin, although it doesn't quite fit the binary intrinsic signature